### DISCUSSION TITLE
After install and registration, I cannot start Firefly (An exception has been thrown during the rendering of a template)

### DISCUSSION THREAD

**Discussion by bandiba** [firefly-iii/firefly-iii#9108] [General]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I managed to install Firefly, could register, and log in. Created a test account etc. However, after a while a Whoops error was displayed, and I could not get into Firefly any more at the defined domain name. 

"Unfortunately, this error was not recoverable :(. Firefly III broke. The error is:

An exception has been thrown during the rendering of a template ("file_put_contents(/var/www/firefly-iii/storage/framework/cache/data/f1/27/f127c8487f61dca9911a93f9ccb47b406acb934c): Failed to open stream: No such file or directory").

This error occured in file /var/www/firefly-iii/resources/views/index.twig on line 108 with code 0."

nginx seems to be OK, I get this displayed at localhost:

"Welcome to nginx!
If you see this page, the nginx web server is successfully installed and working. Further configuration is required.

For online documentation and support please refer to [nginx.org](http://nginx.org/).
Commercial support is available at [nginx.com](http://nginx.com/).

Thank you for using nginx."

Could someone pls advice what I could do.

It is a no-docker installation, basically following this instruction: https://ugeeklab.com/how-to-install-firefly-iii-on-ubuntu-22-04-lts/

### Debug information

I'm sorry I do not know how to do it.

**Comment by JC5** [comment]
Check out these instructions! Let me know if they are clear enough :) 

https://docs.firefly-iii.org/references/faq/docker/#i-get-permission-denied-errors-on-the-cache-folder

**Comment by bandiba** [comment]
Thank you for the quick reply!

I did not use docker for the installation, so I did not execute the docker command. 

I have browsed to the domain/flush page, another error was displayed, but then Firefly started... 

Thank you for the advice!

I'm still experimenting with Firefly, the error page comes up relatively often, at different tasks/pages, so I flush, and carry on... A little bit annoying.

Would the docker based installation be more stable? If yes, do I need to remove/clean the current installation somehow before going for docker?

Thanks!

**Reply by JC5** [reply]
See https://docs.firefly-iii.org/references/faq/install/#unable-to-write-to-cache-directory 

Docker and a normal installation aren't that much different.

**Reply by bandiba** [reply]
Thanks, this helped!
